mlrun/api/crud/model_endpoints.py

```diff
@@ -1034,45 +1190,212 @@ def build_kv_cursor_filter_expression(
-        return "{} {} {} * *".format(
-            cron_trigger["minute"], cron_trigger["hour"], cron_trigger["day"]
-        ).replace("None", "*")
+        # Retrieve the raw data from the time series DB based on the provided metrics and time ranges
+        frames_client = mlrun.utils.v3io_clients.get_frames_client(
+            token=self.access_key,
+            address=mlrun.mlconf.v3io_framesd,
+            container=self.container,
+        )
+
+        try:
+            data = frames_client.read(
+                backend="tsdb",
+                table=events_path,
+                columns=["endpoint_id", *metrics],
+                filter=f"endpoint_id=='{endpoint_id}'",
+                start=start,
+                end=end,
+            )
+
+            # Fill the metrics mapping dictionary with the metric name and values
+            data_dict = data.to_dict()
+            for metric in metrics:
+                metric_data = data_dict.get(metric)
+                if metric_data is None:
+                    continue
+
+                values = [
+                    (str(timestamp), value) for timestamp, value in metric_data.items()
+                ]
+                metrics_mapping[metric] = mlrun.api.schemas.Metric(
+                    name=metric, values=values
+                )
+        except v3io_frames.errors.ReadError:
+            logger.warn("Failed to read tsdb", endpoint=endpoint_id)
+        return metrics_mapping
+
+
+class _ModelEndpointSQLStore(_ModelEndpointStore):
+    @abstractmethod
+    def write_model_endpoint(self, endpoint, update=True):
+        raise NotImplementedError
+
+    @abstractmethod
+    def update_model_endpoint(self, endpoint_id, attributes):
+        raise NotImplementedError
+
+    @abstractmethod
+    def delete_model_endpoint(self, endpoint_id):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_model_endpoint(
+        self,
+        metrics: typing.List[str] = None,
+        start: str = "now-1h",
+        end: str = "now",
+        feature_analysis: bool = False,
+        endpoint_id: str = None,
+        raise_for_status=None,
+    ):
+        raise NotImplementedError
+
+    def list_model_endpoints(
+        self, model: str, function: str, labels: typing.List, top_level: bool
+    ):
+        raise NotImplementedError
+
+
+def get_endpoint_target(project, access_key):
+    """
+    Getting the DB target type based on mlrun.config.model_endpoint_monitoring.store_type
+    :param project:             The name of the project.
+    :param access_key:          Access key with permission to write to DB table.
+
+    :return: ModelEndpointTarget object. Using this object, the user can apply different operations on the
+             model endpoint record such as write, update, get and delete.
+    """
+    if mlrun.mlconf.model_endpoint_monitoring.store_type == "kv":
+        return _ModelEndpointKVStore(project=project, access_key=access_key)
+    else:
```

Comment:
Should be more strict here and define a specific value for SQL (probably `sql` would work). Explode if the value provided is not one of the supported types. Best is to define an enum with the possible values, which can also have an accessor which returns the class mapped to each value.

Reply:
A new enum class `ModelEndpointStoreType` which returns the `_ModelEndpointStore` object (under `to_endpoint_target()`) based on the provided value 
```
class ModelEndpointStoreType(enum.Enum):
    """Enum class to handle the different store type values for saving a model endpoint record."""

    kv = "kv"
    sql = "sql"

    def to_endpoint_target(self, project, access_key):
        """
        Return a ModelEndpointTarget object based on the provided enum value.

        :param project:             The name of the project.
        :param access_key:          Access key with permission to write to DB table.

        :return: ModelEndpointTarget object.
        """

        if self.value == ModelEndpointStoreType.kv.value:
            return _ModelEndpointKVStore(project=project, access_key=access_key)
        elif self.value == ModelEndpointStoreType.sql.value:
            return _ModelEndpointSQLStore(project=project, access_key=access_key)

    @classmethod
    def _missing_(cls, value: typing.Any):
        """A lookup function to handle an invalid value.

        :param value: Provided enum (invalid) value.

        """
        valid_values = list(cls.__members__.keys())
        raise mlrun.errors.MLRunInvalidArgumentError(
            "%r is not a valid %s, please choose a valid value: %s."
            % (value, cls.__name__, valid_values)
        )
```